tests/vnc_test.go

```diff
@@ -155,6 +161,54 @@ var _ = Describe("[rfe_id:127][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			_, err = wrappedRoundTripper.RoundTrip(req)
 			Expect(err).ToNot(HaveOccurred())
 		})
+
+		It("should connect to vnc with --proxy-only flag", func() {
+			By("Invoking virtctl vnc with --proxy-only")
+			proxyOnlyCommand := tests.NewVirtctlCommand("vnc", "--proxy-only", "--namespace", vmi.Namespace, vmi.Name)
+
+			r, w, _ := os.Pipe()
```

Comment:
This creates a pipe on the OS level. It may be simpler to specify an `io.Writer` as the output of the `proxyOnlyCommand`. Something like:

```go
pipeReader, pipeWriter := io.Pipe()
proxyOnlyCommand.SetOut(pipeWriter)
go func() {
	defer GinkgoRecover()
	Expect(proxyOnlyCommand.Execute()).ToNot(HaveOccurred())
}()

var result map[string]interface{}
err := json.NewDecoder(pipeReader).Decode(&result)
Expect(err).ToNot(HaveOccurred())

// Discards the rest of the output from the pipe, so the command does not block
go io.Copy(ioutil.Discard, pipeReader)
```

Also this way there is no need to call `time.Sleep()`, the `Decode()` method should block until some data is available.

Reply:
Ignore my previous comment. I incorrectly thought that `proxyOnlyCommand` would run `virtctl` as a new process.
The current approach with redirecting `os.Stdout` is ok.

Reply:
Corrected